component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteStore.java

```diff
@@ -0,0 +1,1158 @@
+            ApacheIgniteConditionVisitor visitor = new ApacheIgniteConditionVisitor();
+            orderByAttributeBuilder.getExpressionBuilder().build(visitor);
+            String compiledCondition = visitor.returnCondition();
+            compiledSelectionList.append(compiledCondition);
+            OrderByAttribute.Order order = orderByAttributeBuilder.getOrder();
+            if (order == null) {
+                compiledSelectionList.append(SEPARATOR);
+            } else {
+                compiledSelectionList.append(WHITESPACE).append(order.toString()).append(SEPARATOR);
+            }
+            Map<Integer, Object> conditionParamMap = visitor.getParameters();
+            int maxOrdinal = 0;
+            for (Map.Entry<Integer, Object> entry : conditionParamMap.entrySet()) {
+                Integer ordinal = entry.getKey();
+                paramMap.put(ordinal + offset, entry.getValue());
+                if (ordinal > maxOrdinal) {
+                    maxOrdinal = ordinal;
+                }
+            }
+            offset = maxOrdinal;
+        }
+        if (compiledSelectionList.length() > 0) {
+            compiledSelectionList.setLength(compiledSelectionList.length() - 1);
+        }
+        return new ApacheIgniteCompiledCondition(compiledSelectionList.toString(), paramMap, null);
+    }
+
+    /**
+     * Create a table in ignite store if it does not exists already
+     *
+     * @param primaryKey primary keys of the table to be created.
+     */
+    private void createTable(Annotation primaryKey) throws ConnectionUnavailableException {
+        PreparedStatement statement = null;
+        try {
+            StringBuilder tableCreateQuery = new StringBuilder();
+            List<Element> primaryKeyList = (primaryKey == null) ? new ArrayList<>() : primaryKey.getElements();
+            tableCreateQuery.append(TABLE_CREATE_QUERY).append(this.tableName).append(OPEN_PARENTHESIS);
+            this.attributes.forEach(attribute -> {
+                tableCreateQuery.append(attribute.getName()).append(WHITESPACE);
+                switch (attribute.getType()) {
+                    case BOOL:
+                        tableCreateQuery.append(ApacheIgniteConstants.BOOLEAN).append(WHITESPACE);
+                        break;
+                    case DOUBLE:
+                        tableCreateQuery.append(ApacheIgniteConstants.DOUBLE).append(WHITESPACE);
+                        break;
+                    case FLOAT:
+                        tableCreateQuery.append(ApacheIgniteConstants.FLOAT).append(WHITESPACE);
+                        break;
+                    case INT:
+                        tableCreateQuery.append(ApacheIgniteConstants.INTEGER).append(WHITESPACE);
+                        break;
+                    case LONG:
+                        tableCreateQuery.append(ApacheIgniteConstants.LONG).append(WHITESPACE);
+                        break;
+                    case STRING:
```

Comment:
If this switch case for siddhi data types. then it can also include OBJECT case. This should be handled.